cpp/src/io/csv/type_conversion.cuh

```diff
@@ -332,8 +200,94 @@ int32_t convertStrtoHash(const char * key, long start_idx, long end_idx, uint32_
     return h1;
 }
 
+__host__ __device__ gdf_date32 parseDateFormat(char *data, long start_idx, long end_idx, bool dayfirst);
+__host__ __device__ gdf_date64 parseDateTimeFormat(char *data, long start_idx, long end_idx, bool dayfirst);
+
+typedef struct parsing_opts_ {
+	char				delimiter;
+	char				terminator;
+	char				quotechar;
+	bool				keepquotes;
+	bool				dayfirst;
+	char				decimal;
+	char				thousands;
+	int32_t*			trueValues;
+	int32_t*			falseValues;
+	int32_t				trueValuesCount;
+	int32_t				falseValuesCount;
+} parsing_opts_t;
+
+// Default function for parsing and extracting a GDF value from a CSV field
+// Handles all integer and floating point data types
+// Other GDF data types are templated for their individual, unique parsing
+template<typename T>
+__host__ __device__
+T convertStrToValue(const char *data, long start, long end, const parsing_opts_t& opts) {
+
+	T value = 0;
+
+	// Handle negative values if necessary
+	int32_t sign = 1;
+	if (data[start] == '-') {
+		sign = -1;
+		start++;
+	}
+
+	// Handle the whole part of the number
+	long index = start;
+	while (index <= end) {
+		if (data[index] == opts.decimal) {
+			++index;
+			break;
+		}
+		else if (data[index] != opts.thousands) {
+			value *= 10;
+			value += data[index] -'0';
+		}
+		++index;
+	}
+
+	// Handle fractional part of the number if necessary
+	int32_t divisor = 1;
+	while (index <= end) {
+		if (data[index] != opts.thousands) {
+			value *= 10;
+			value += data[index] -'0';
+			divisor *= 10;
+		}
+		++index;
+	}
+
+	return (divisor > 1) ? (value * sign / divisor) : (value * sign);
+}
+
+template<>
+__host__ __device__
+cudf::date32 convertStrToValue<cudf::date32>(const char *data, long start, long end, const parsing_opts_t& opts) {
+
+	return cudf::date32{ parseDateFormat(const_cast<char*>(data), start, end, opts.dayfirst) };
```

Comment:
Would it be more appropriate to make the `data` parameter for `parseDateFormat` to be `const` rather than having to use `const_cast`. 

Based on a cursory examination, it does not appear that `parseDateFormat` should be able to modify `data`, and therefore for `const` correctness, should be marked as `const`.

Reply:
Changed the parameters to be const char* in the child functions. Thanks!